oqs/wrapper.py

```diff
@@ -146,9 +119,16 @@ def __init__(self, alg_name, secret_key = None):
             'length_secret_key' : int(self._kem.contents.length_secret_key),
             'length_ciphertext' : int(self._kem.contents.length_ciphertext),
             'length_shared_secret' : int(self._kem.contents.length_shared_secret) }
+
         if secret_key:
             self.secret_key = ct.create_string_buffer(secret_key, self._kem.contents.length_secret_key)
 
+    def __enter__(self):
+        return self
+
+    def __exit__(self, ctx_type, ctx_value, ctx_traceback):
+        self.free()
+
     def generate_keypair(self):
```

Comment:
Rename generate\_keypair to keypair. Similarly for encaps and decaps.

Reply:
The keypair/encaps/decaps functions are very unclear. The OQS C API stays close to the NIST API, sure, but I believe the higher-level languages should aim to be clearer. Python programmers shouldn't have to be familiar with the NIST names. Keypair by itself is cryptic; is it to obtain the key, set it, generate it?

Reply:
Python favours being explicit & readability. A simple solution is to add something like "generate_keypair() maps to keypair() in the C code" in the method docstring.

Reply:
Basic docstrings have now been added to outline the mapping.